### PR TITLE
test: cache global template build at initialization

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

Faster tests

## Solution

Build only once at global template initialization rather than per each test.

This requires that we also use a single Solc version for all tests, at least by default. Do that.

### CODE REVIEW

File: crates/test-utils/src/util.rs [hunk 1]
```diff
@@ -51,30 +53,62 @@ pub fn template_lock() -> RwLock<File> {
...
-        drop(read);
+        // We are the first to acquire the lock:
+        // - initialize a new empty temp project;
+        // - run `forge init`;
+        // - run `forge build`;
+        // - copy it over to the global template;
+        // Ideally we would be able to initialize a temp project directly in the global template,
+        // but `TempProject` does not currently allow this: https://github.com/foundry-rs/compilers/issues/22
+
+        // Release the read lock and acquire a write lock, initializing the lock file.
```
Comment (mattsse) [inline crates/test-utils/src/util.rs:87]: ah yeah we should have thought about this much earlier lol

**Review by mattsse — APPROVED** [review]

File: crates/test-utils/src/util.rs (hunk 1 above)
Comment (DaniPopes) [reply crates/test-utils/src/util.rs:87]: The problem is that the static Lazy didn't work due to nextest running each test in a separate process

**Comment by DaniPopes** [comment]
Solidity version used:
- 0.5.17
- 0.7.1
- 0.8.10
- 0.8.13
- 0.8.15
- 0.8.17
- 0.8.23

Now down to:
- 0.5.17 (`can_test_pre_bytecode_hash`)
- 0.8.22
- 0.8.23

**Review by Evalir — APPROVED** [review]
this is sick. the note about the global lock makes a ton of sense

**Pull request merged**